containers/src/Kokkos_DualView.hpp

```diff
@@ -942,24 +960,50 @@ class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
     if (modified_flags(1) >= modified_flags(0)) {
       /* Resize on Device */
       if (sizeMismatch) {
-        ::Kokkos::resize(d_view, n0, n1, n2, n3, n4, n5, n6, n7);
-        h_view = create_mirror_view(d_view);
+        ::Kokkos::resize(arg_prop..., d_view, n0, n1, n2, n3, n4, n5, n6, n7);
+        h_view = create_mirror_view(typename t_host::memory_space(), d_view,
+                                    arg_prop...);
 
         /* Mark Device copy as modified */
         modified_flags(1) = modified_flags(1) + 1;
       }
     } else {
       /* Realloc on Device */
       if (sizeMismatch) {
-        ::Kokkos::resize(h_view, n0, n1, n2, n3, n4, n5, n6, n7);
-        d_view = create_mirror_view(typename t_dev::execution_space(), h_view);
+        ::Kokkos::resize(arg_prop..., h_view, n0, n1, n2, n3, n4, n5, n6, n7);
+        d_view = create_mirror_view(typename t_dev::memory_space(), h_view,
```

Comment:
Why the change from exec space to memory space?

Reply:
IMHO, `create_mirror_view*` should always be called with a memory space since we want to specify where the View that we return is allocated and not where any kind of initialization/copy is performed. This probably only matters for the host pinned spaces, though.
I'm happy to change it back if there is a good reason for specifying an execution space that I missed.

Reply:
We should bring it to the group. If something in Kokkos makes no sense, it's either

1) Something meaningless
2) Something direly important that will shake the world to its foundations if changed ;)

Reply:
I think I agree with Daniel (note didn't do a full review)